package/kernel/mac80211/patches/ath9k/531-ath9k_extra_platform_leds.patch

```diff
@@ -33,30 +25,30 @@
  void ath_deinit_leds(struct ath_softc *sc)
  {
  	struct ath_led *led;
-@@ -124,8 +143,10 @@ void ath_deinit_leds(struct ath_softc *s
+@@ -124,8 +142,10 @@ void ath_deinit_leds(struct ath_softc *s
  
  void ath_init_leds(struct ath_softc *sc)
  {
-+	struct ath9k_platform_data *pdata = sc->dev->platform_data;
++	struct ath_hw *ah = sc->sc_ah;
  	char led_name[32];
  	const char *trigger;
 +	int i;
  
  	INIT_LIST_HEAD(&sc->leds);
  
-@@ -134,6 +155,17 @@ void ath_init_leds(struct ath_softc *sc)
+@@ -134,6 +154,17 @@ void ath_init_leds(struct ath_softc *sc)
  
  	ath_fill_led_pin(sc);
  
-+	if (pdata && pdata->leds && pdata->num_leds)
-+		for (i = 0; i < pdata->num_leds; i++) {
-+			if (pdata->leds[i].gpio == sc->sc_ah->led_pin)
-+				sc->sc_ah->led_pin = -1;
++	if (ah->leds && ah->num_leds)
```

Comment:
This variable `num_leds` seems always be 0.

Reply:
I'm guessing this worked with platform data. No op with OF.